web/src/app/projects/repository/repository.component.html

```diff
@@ -10,30 +10,144 @@
+              <p fxFlex="85%" class="bottom-line" mat-line>Total commits : 136</p>
+            </div>
+          </mat-card-actions>
+        </mat-card>
+      </div>
+      <div fxFlex="24%">
+        <mat-card>
+          <mat-card-content class="dashboard-card">
+            <div class="card-icon info-outline">
+              <mat-icon>control_point</mat-icon>
+            </div>
+            <p class="heading">Latest PR</p>
+            <h3 class="description">1347</h3>
+          </mat-card-content>
+          <mat-card-actions>
+            <div class="card-footer" fxLayout="row">
+              <img fxFlex="15%" alt ="webkhushboo" src="https://avatars2.githubusercontent.com/u/10898595?v=4"/>
+              <p fxFlex="85%" class="bottom-line" mat-line>Found a bug | webkhushboo</p>
+            </div>
+          </mat-card-actions>
+        </mat-card>
+      </div>
+    </div>
+    <div fxLayout="row" fxLayoutGap="20px" class="tab-list">
+      <div fxFlex="24%">
+        <mat-card>
+          <mat-card-content class="dashboard-card">
+            <h3 *ngIf="repository.events && repository.events.length > 0; else eventsOther" class="heading card-icon">Events List</h3>
+            <ng-template #eventsOther><p> No events are available</p></ng-template>
+            <mat-list>
+              <mat-list-item *ngFor="let event of repository.events | slice:0:5">
+                <img matListAvatar [src]="event.actor.avatarUrl" [alt]="event.actor.username" />
+                <h4 mat-line [matTooltip]="event.payload.title">{{ event.payload.title }}</h4>
+                <p mat-line>{{event.type}}</p>
+                <p mat-line>{{ event.createdOn | timeAgo }}</p>
+              </mat-list-item>
+            </mat-list>
+          </mat-card-content>
+        </mat-card>
+      </div>
+      <div fxFlex="24%">
+        <mat-card>
+          <mat-card-content class="dashboard-card">
+            <h3 *ngIf="repository.issues && repository.issues.length > 0; else issuesOther" class="heading card-icon control-point">Issues</h3>
+            <ng-template #issuesOther><p> No issues are available</p></ng-template>
+            <mat-list role="list">
+              <mat-list-item role="listitem" *ngFor="let issue of repository.issues | slice:0:5">
+                <img matListAvatar [src]="issue.owner.avatarUrl" [alt]="issue.owner.username" />
+                <h4 mat-line><a href="{{issue.url}}">
+                    #{{issue.number}} </a> {{issue.title}}</h4>
+                    <p mat-line>Last updated: {{issue.updatedOn | timeAgo}}</p>
+              </mat-list-item>
+            </mat-list>
+          </mat-card-content>
+        </mat-card>
+      </div>
+      <div fxFlex="24%">
+        <mat-card>
+          <mat-card-content class="dashboard-card">
+            <h3 *ngIf="repository.contributors && repository.contributors.length > 0; else contributorsOther" class="heading card-icon verified-user">Contributors</h3>
```

Comment:
Can the `*ngIf` be lower down, so the heading always displays.

I think this `repository.contributors.length > 0` can just be `repository.contributors.length`.

Reply:
Can we know the reason for not using else ? I don't mind using another *ngIf but thats not good coding I guess.

Reply:
I think it reads cleaner, because each section is independent and clear what the condition is - just my personal opinion though 🤓

Reply:
but the processing will be slow to checks two ifs , instead checking one if then going to else if condition not matches

Reply:
Use two if statements if both if statement conditions could be true at the same time. In this example, both conditions can be true. You can pass and do great at the same time. Use an if/else statement if the two conditions are mutually exclusive meaning if one condition is true the other condition must be false.

Reply:
Yes you are right 👍